src/arkode/arkode_sprkstep.c

```diff
@@ -0,0 +1,781 @@
+  {
+    sunrealtype ai = step_mem->method->a[is];
+    sunrealtype bi = step_mem->method->b[is];
+
+    /* Set current stage time(s) */
+    ark_mem->tcur = ark_mem->tn + ai * ark_mem->h;
+
+    /* Evaluate p' with the previous velocity */
+    N_VConst(ZERO, step_mem->sdata); /* either have to do this or ask user to
+                                        set other outputs to zero */
+    retval = sprkStep_f1(step_mem, ark_mem->tcur, prev_stage, step_mem->sdata,
+                         ark_mem->user_data);
+    if (retval != 0) { return ARK_RHSFUNC_FAIL; }
+
+    /* Position update */
+    N_VLinearSum(ONE, prev_stage, ark_mem->h * bi, step_mem->sdata, curr_stage);
+
+    /* Evaluate q' with the current positions */
+    N_VConst(ZERO, step_mem->sdata); /* either have to do this or ask user to
+                                        set other outputs to zero */
+    retval = sprkStep_f2(step_mem, ark_mem->tcur, curr_stage, step_mem->sdata,
+                         ark_mem->user_data);
+    if (retval != 0) { return ARK_RHSFUNC_FAIL; }
+
+    /* Velocity update */
+    N_VLinearSum(ONE, curr_stage, ark_mem->h * ai, step_mem->sdata, curr_stage);
+
+    /* apply user-supplied stage postprocessing function (if supplied) */
+    if (ark_mem->ProcessStage != NULL)
+    {
+      retval = ark_mem->ProcessStage(ark_mem->tcur, ark_mem->ycur,
+                                     ark_mem->user_data);
+      if (retval != 0) { return (ARK_POSTPROCESS_STAGE_FAIL); }
+    }
+
+    /* keep track of the stage number */
+    step_mem->istage++;
+
+    prev_stage = curr_stage;
+  }
+
+  *nflagPtr = 0;
+  *dsmPtr   = 0;
+
+  return ARK_SUCCESS;
+}
+
+/* Increment SPRK algorithm with compensated summation.
+   This algorithm requires 6 vectors, but 5 of them are reused
+   from the ARKODE core. */
+int sprkStep_TakeStep_Compensated(void* arkode_mem, realtype* dsmPtr,
+                                  int* nflagPtr)
+{
+  ARKodeMem ark_mem          = NULL;
+  ARKodeSPRKStepMem step_mem = NULL;
+  ARKodeSPRKStorage method   = NULL;
+  int retval                 = 0;
+  int is                     = 0;
+  N_Vector delta_Yi          = NULL;
+  N_Vector yn_plus_delta_Yi  = NULL;
+  N_Vector diff              = NULL;
+
+  /* access ARKodeSPRKStepMem structure */
+  retval = sprkStep_AccessStepMem(arkode_mem, "sprkStep_TakeStep_SPRK",
+                                  &ark_mem, &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  method = step_mem->method;
+
+  /* Vector shortcuts */
+  delta_Yi         = ark_mem->tempv1;
+  yn_plus_delta_Yi = ark_mem->tempv2;
+  diff             = ark_mem->tempv3;
+
+  /* [ \Delta Q_0 ] = [ 0 ]
+     [ \Delta P_0 ] = [ 0 ] */
+  N_VConst(ZERO, delta_Yi);
+
+  /* loop over internal stages to the step */
+  for (is = 0; is < method->stages; is++)
+  {
+    /* store current stage index */
+    step_mem->istage = is;
+
+    /* set current stage time(s) */
+    ark_mem->tcur = ark_mem->tn + method->b[is] * ark_mem->h;
```

Comment:
This uses b for the abscissa where before it used a. Does this matter?

Reply:
There was an inconsistency, although it doesnt matter for the problems we currently support.